napari/_qt/qt_main_window.py

```diff
@@ -1730,6 +1731,67 @@ def export_figure(
             imsave(path, img)
         return img
 
+    def export_rois(
+        self,
+        rois: np.ndarray,
+        paths: list[str] | None = None,
+        scale: float | None = None,
+    ):
+        """Export the shapes rois with storage file paths.
+
+        Parameters
+        ----------
+        rois: np.ndarray
+            An array of shape (n, 2, 2) where n is the number of rois
+            and the first two coordinates correspond to the top left and
+            the last two coordinates correspond to the bottom right corners
+        paths: list
+            The list to store file path for shapes roi
+
+        Returns
+        -------
+        screenshot_list: list
+            The list with roi screenshots.
+
+        """
+        if paths is not None and len(paths) != len(rois):
+            raise ValueError(
+                trans._(
+                    'The number of file paths does not match the number of ROI shapes',
+                    deferred=True,
+                )
+            )
+
+        screenshot_list = []
+        camera = self._qt_viewer.viewer.camera
+        start_camera_center = camera.center
+        start_camera_zoom = camera.zoom
+        canvas = self._qt_viewer.canvas
+        prev_size = canvas.size
+
+        for index, shape in enumerate(rois):
+            top_left = shape[0]
+            bottom_right = shape[1]
+            x1, y1 = top_left
+            x2, y2 = bottom_right
+            if x1 > x2 or y1 > y2:
+                raise ValueError('ROI shape error')
+            center_coord, height, width = get_center_bbox(shape)
+            camera.center = center_coord
+            canvas.size = (int(height), int(width))
+
+            camera.zoom = 1.0
+            path = paths[index] if paths is not None else None
+            screenshot_list.append(
+                self.screenshot(path=path, canvas_only=True, scale=scale)
+            )
+
+            canvas.size = prev_size
+            camera.center = start_camera_center
+            camera.zoom = start_camera_zoom
```

Comment:
Probably better to de-dent this part right?

```suggestion
        canvas.size = prev_size
        camera.center = start_camera_center
        camera.zoom = start_camera_zoom
```